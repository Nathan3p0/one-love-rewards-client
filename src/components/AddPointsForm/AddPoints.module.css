.points__update {
    width: 90%;
    height: 100vh;
}

.points__update_purchase,
.points__update_current,
.points__update_new {
    height: 10rem;
    width: 15.6rem;
    background-color: #429398;
    box-shadow: 1px 2px 3px #000;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    color: #fff;
}

.points__update_purchase p,
.points__update_current p,
.points__update_new p {
    font-size: 2.5rem;
    margin: 0;
}

.points__update_purchase h3,
.points__update_current h3,
.points__update_new h3 {
    font-size: 1.5rem;
    margin: 1rem 0 0 0;
    color: #eee;
    letter-spacing: .1rem;
}

.points__update_totals {
    margin-top: 1rem;
    display: grid;
    /* grid-gap: 1rem; */
    grid-template-columns: 1fr 1fr 1fr;
}

.points__update_formcontainer {
    margin-top: 1rem;
    display: grid;
    grid-template-columns: 1fr 1fr;
    /* width: 50rem; */
    grid-gap: 1rem;
}

.points__update_formcontainer h4 {
    font-size: 1.2rem;
    margin-top: 1rem;
    color: #6DDB1F;
    font-weight: 700;
    text-shadow: 1px 1px 3px black;
}

.points__update_formcontainer p {
    font-size: 1.2rem;
    margin: .5rem;
    color: #FDFFFC;
    font-weight: 700;
    text-shadow: 1px 1px 3px black;
}

.points__update_form input[type="text"] {
    width: 100%;
    border: .2rem solid rgba(0, 0, 0, 0.3);
    border-radius: .3rem;
    font-size: 1.5em;
    margin-right: .2rem;
    margin-top: 1rem;
    padding: .8rem 0;
}

.points__update_form {
    margin-top: 1.5rem;
}

.points__update_form label {
    color: #FDFFFC;
}

button.done_btn {
    width: 15rem;
    margin-top: 2rem;
    transition: all 0.5s cubic-bezier(0.65, -0.25, 0.25, 1.95);
    font-weight: 900;
    color: #202121;
    padding: 1.25rem 2rem;
    background: #FFE74C;
    text-transform: uppercase;
}

button.done_btn:hover,
button.done_btn:focus,
button.done_btn:active {
    letter-spacing: 0.125rem;
}

.rewards_list {
    text-align: center;
    max-width: 100%;
}

.rewards_list ul {
    list-style-type: none;
}

.rewards_list h3 {
    /* margin-top: 2rem; */
    font-size: 1.7rem;
    color: #FDFFFC;
}

@media screen and (max-width: 1200px) {
    .points__update {
        width: 90%;
    }

    .points__update_totals {
        grid-template-columns: 1fr 1fr 1fr;
    }

    .points__update_formcontainer {
        margin-top: 1rem;
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-gap: 1rem;
    }

    .points__update_purchase,
    .points__update_current,
    .points__update_new {
    height: 8rem;
    width: 14rem;
    background-color: rgb(75, 89, 102);
    box-shadow: 1px 2px 3px #000;
    color: #fff;
}
}

@media screen and (max-width: 720px) {
    .points__update_totals {
        grid-template-columns: 1fr;
    }
    .points__update_purchase,
    .points__update_current,
    .points__update_new {
    width: 100%;
    }

    .points__update_formcontainer {
        grid-template-columns: 1fr;
    }

    .points__update {
        background: #253034;
        height: 100%;
        width: 90%;
    }
}

@media screen and (max-width: 320px) {
    .points__update {
        width: 100%;
        padding-right: 0 1rem;
    }

    .points__update_formcontainer {
        padding: 1rem;
    }

    .rewards_list ul {
        padding-left: 0;
    }

    .points__update_totals { padding: 1rem;
    }
}