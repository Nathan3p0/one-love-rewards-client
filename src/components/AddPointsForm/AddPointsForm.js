import React, { Component } from 'react'
import styles from './AddPoints.module.css'
import RewardsListItem from '../RewardsListItem/RewardsListItem'

class AddPointsForm extends Component {

    componentDidMount() {
        this.props.fetchCustomerPointsInfo(this.props.id)
        this.props.fetchRewards()
    }

    render() {
        let rewardsItems
        if (this.props.rewards === undefined) {
            rewardsItems = []
        } else {
            rewardsItems = this.props.rewards.map((item, i) =>
        <RewardsListItem key={i} currentPoints={this.props.points} pointsRequired={item.points_required} discount={item.discount} onClickReward={this.props.onClickReward} updated={this.props.updated} />
            )}
        return (
            <section className={styles.points__update}>
                <div className={styles.points__update_totals}>
                    <div className={styles.points__update_current}>
                        <p>{this.props.points}pts</p>
                        <h3>Account Balance</h3>
                    </div>
                    <div className={styles.points__update_new}>
                        <p>{this.props.updated}pts</p>
                        <h3>Updated Balance</h3>
                    </div>
                    <div className={styles.points__update_purchase}>
                        <p>${this.props.purchase}</p>
                        <h3>Today's Purchase</h3>
                    </div>
                </div>
                <div className={styles.points__update_formcontainer}>
                    <div>
                        <div role='alert'>
                            {this.props.error && <p className='red'>{this.props.error}</p>}
                        </div>
                        <p>{this.props.name}</p>
                        <p>{this.props.phone}</p>
                        <p>{this.props.email}</p>
                        {this.props.selected && <h4>{this.props.selected} to be redeemed.</h4>}
                        <form onSubmit={this.props.handlePointsSubmit} className={styles.points__update_form}>
                            <label htmlFor="purchase-total">
                                Purchase Total:
                            </label><br />
                            <input type="number" id="purchase-total" name="purchase_total" value={this.props.purchase} onChange={this.props.handleInputChange} required /><br />
                            <button className={styles.done_btn} type="submit">Done</button>
                        </form>
                    </div>
                    <div className={styles.rewards_list}>
                        <h3>Rewards Available</h3>
                        <ul>
                            {rewardsItems}
                        </ul>
                    </div>
                </div>
            </section>
        )
    }
}

export default AddPointsForm