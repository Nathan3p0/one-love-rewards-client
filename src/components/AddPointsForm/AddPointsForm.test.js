import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import AddPointsForm from './AddPointsForm'

describe('Add Points Form Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<AddPointsForm />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<AddPointsForm />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<AddPointsForm
            points={0}
            id={12}
            handleInputChange={() => {}}
            purchase={12}
            updated={24}
            name={'Nathan'}
            email={'nszelag@live.com'}
            phone={'734\'673\'5101'}
            handlePointsSubmit={() => {}}
            fetchRewards={() => {}}
            rewards={[]}
            error={'error'}
            onClickReward={() => {}}
            selected={'$5 Off'} />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})