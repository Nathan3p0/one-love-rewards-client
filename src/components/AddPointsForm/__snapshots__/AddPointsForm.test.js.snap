// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Points Form Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="points__update"
>
  <div
    className="points__update_totals"
  >
    <div
      className="points__update_current"
    >
      <p>
        pts
      </p>
      <h3>
        Account Balance
      </h3>
    </div>
    <div
      className="points__update_new"
    >
      <p>
        pts
      </p>
      <h3>
        Updated Balance
      </h3>
    </div>
    <div
      className="points__update_purchase"
    >
      <p>
        $
      </p>
      <h3>
        Today's Purchase
      </h3>
    </div>
  </div>
  <div
    className="points__update_formcontainer"
  >
    <div>
      <div
        role="alert"
      />
      <p />
      <p />
      <p />
      <form
        className="points__update_form"
      >
        <label
          htmlFor="purchase-total"
        >
          Purchase Total:
        </label>
        <br />
        <input
          id="purchase-total"
          name="purchase_total"
          placeholder="Purchase Total"
          required={true}
          type="text"
        />
        <br />
        <button
          className="done_btn"
          type="submit"
        >
          Done
        </button>
      </form>
    </div>
    <div
      className="rewards_list"
    >
      <h3>
        Rewards Available
      </h3>
      <ul />
    </div>
  </div>
</section>
`;

exports[`Add Points Form Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<section
  className="points__update"
>
  <div
    className="points__update_totals"
  >
    <div
      className="points__update_current"
    >
      <p>
        0
        pts
      </p>
      <h3>
        Account Balance
      </h3>
    </div>
    <div
      className="points__update_new"
    >
      <p>
        24
        pts
      </p>
      <h3>
        Updated Balance
      </h3>
    </div>
    <div
      className="points__update_purchase"
    >
      <p>
        $
        12
      </p>
      <h3>
        Today's Purchase
      </h3>
    </div>
  </div>
  <div
    className="points__update_formcontainer"
  >
    <div>
      <div
        role="alert"
      >
        <p
          className="red"
        >
          error
        </p>
      </div>
      <p>
        Nathan
      </p>
      <p>
        734'673'5101
      </p>
      <p>
        nszelag@live.com
      </p>
      <h4>
        $5 Off
         to be redeemed.
      </h4>
      <form
        className="points__update_form"
        onSubmit={[Function]}
      >
        <label
          htmlFor="purchase-total"
        >
          Purchase Total:
        </label>
        <br />
        <input
          id="purchase-total"
          name="purchase_total"
          onChange={[Function]}
          placeholder="Purchase Total"
          required={true}
          type="text"
          value={12}
        />
        <br />
        <button
          className="done_btn"
          type="submit"
        >
          Done
        </button>
      </form>
    </div>
    <div
      className="rewards_list"
    >
      <h3>
        Rewards Available
      </h3>
      <ul />
    </div>
  </div>
</section>
`;
